external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -484,8 +484,11 @@ private boolean emitOrRetryTuple(ConsumerRecord<K, V> record) {
                     return true;
                 }
             } else {
+                /*if a null tuple is not configured to be emitted, it should be marked as emitted and acked immediately
+                * to allow its offset to be commited to Kafka*/
                 LOG.debug("Not emitting null tuple for record [{}] as defined in configuration.", record);
                 msgId.setEmitted(false);
```

Comment:
I'd like to see this property renamed to "isFiltered" or "isNullTuple" something similar. Right now we have both a list of emitted tuples, and also the isEmitted property, and they don't always match. It seems unnecessarily confusing.

Reply:
I concur. I'll refactor it.